File: test21_time.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <regex.h>
#include <libreplace.h>

#include <time.h>

int main(void){
  
  char *str;
  char *regex = "v[i!1][a@]gr[a@]";
  char *replacement = "viagra";
  regex_t reg;
  clock_t c1,c2;
  int i = 0;

  c1 = clock();
  regcomp(&reg,regex,REG_EXTENDED);
  while(i < 1000000){
    str = (char *)malloc(sizeof(char)*55);
    strcpy(str,"vi@gra v1agra v1@gra v!@gra");
    replaceAll(str,reg,replacement);
    i++;
  }
  c2 = clock();

  printf("%d %d %d\n",c1,c2,CLOCKS_PER_SEC);

  free(str);
  regfree(&reg);
  return 0;
}

File: test21_time.rb
require 'benchmark'

result = Benchmark.realtime do
  num = 0
  while num < 1000000 do
    str = "vi@gra v1agra v1@gra v!@gra"
    str = str.gsub(/v[i!1][a@]gr[a@]/, 'viagra')
    num = num + 1
  end
end
puts "#{result}s"
